tensorboard/webapp/metrics/store/metrics_reducers_test.ts

```diff
@@ -3290,53 +3296,83 @@ describe('metrics reducers', () => {
   });
 
   describe('step selector features', () => {
-    it('toggles whether stepSelector is enabled or not', () => {
+    it('does not enable and disable stepSelection when toggled by fobs', () => {
       const state1 = buildMetricsState({
         stepSelectorEnabled: false,
+        linkedTimeEnabled: false,
       });
 
-      const state2 = reducers(state1, actions.stepSelectorToggled({}));
-      expect(state2.stepSelectorEnabled).toBe(true);
-
-      const state3 = reducers(state2, actions.stepSelectorToggled({}));
-      expect(state3.stepSelectorEnabled).toBe(false);
+      const state2 = reducers(
+        state1,
+        actions.stepSelectorToggled({
+          affordance: TimeSelectionToggleAffordance.FOB_DESELECT,
+        })
+      );
+      expect(state2.stepSelectorEnabled).toBe(false);
     });
 
-    it('disables linkedTime when stepSelector is disabled', () => {
+    it('enables and disables stepSelection when toggled by check box', () => {
       const state1 = buildMetricsState({
         stepSelectorEnabled: false,
         linkedTimeEnabled: false,
       });
 
-      const state2 = reducers(state1, actions.stepSelectorToggled({}));
+      const state2 = reducers(
+        state1,
+        actions.stepSelectorToggled({
+          affordance: TimeSelectionToggleAffordance.CHECK_BOX,
+        })
+      );
       expect(state2.stepSelectorEnabled).toBe(true);
-      expect(state2.linkedTimeEnabled).toBe(false);
 
-      const state3 = buildMetricsState({
-        stepSelectorEnabled: true,
+      const state3 = reducers(
+        state2,
+        actions.stepSelectorToggled({
+          affordance: TimeSelectionToggleAffordance.CHECK_BOX,
+        })
+      );
+      expect(state3.stepSelectorEnabled).toBe(false);
+    });
+
+    it('enables and disables stepSelection when linked time on', () => {
+      const state1 = buildMetricsState({
+        stepSelectorEnabled: false,
         linkedTimeEnabled: true,
       });
-      const state4 = reducers(state3, actions.stepSelectorToggled({}));
-      expect(state4.stepSelectorEnabled).toBe(false);
-      expect(state2.linkedTimeEnabled).toBe(false);
+
+      const state2 = reducers(
+        state1,
+        actions.stepSelectorToggled({
+          affordance: TimeSelectionToggleAffordance.FOB_DESELECT,
+        })
+      );
+      expect(state2.stepSelectorEnabled).toBe(true);
+
+      const state3 = reducers(
+        state2,
+        actions.stepSelectorToggled({
+          affordance: TimeSelectionToggleAffordance.FOB_DESELECT,
+        })
+      );
+      expect(state3.stepSelectorEnabled).toBe(false);
     });
 
-    it('disables rangeSelection when stepSelector is disabled', () => {
+    it('disables linkedTime and rangeSelection when stepSelector is toggled off', () => {
       const state1 = buildMetricsState({
         stepSelectorEnabled: false,
```

Comment:
Again, how can we have linkedTime AND rangeSelection enabled but stepSelector disabled?